docs/tutorials/gluon/model_parallelism.md

```diff
@@ -0,0 +1,373 @@
+val_data = batchify(corpus.valid, args_batch_size)
+test_data = batchify(corpus.test, args_batch_size)
+```
+
+In this example we will create a stacked LSTM with two layers on two GPUs.
+
+
+```python
+num_layers_list = [1, 1]
+ctx_begin = mx.gpu(0)
+ctx_end = mx.gpu(len(num_layers_list) - 1)
+```
+
+
+```python
+ntokens = len(corpus.dictionary)
+
+model = LSTMModel(ntokens, args_emsize, args_nhid, num_layers_list, args_dropout)
+model.init_params()
+model.init_trainer('adadelta', {'learning_rate': args_lr})
+
+loss = gluon.loss.SoftmaxCrossEntropyLoss()
+```
+
+
+```python
+def get_batch(source, i):
+    seq_len = min(args_bptt, source.shape[0] - 1 - i)
+    data = source[i : i + seq_len]
+    target = source[i + 1 : i + 1 + seq_len]
+    return data, target.reshape((-1,))
+
+def detach(hidden):
+    if isinstance(hidden, (tuple, list)):
+        hidden = [detach(i) for i in hidden]
+    else:
+        hidden = hidden.detach()
+    return hidden
+```
+
+
+```python
+def eval(data_source):
+    total_L = 0.0
+    ntotal = 0
+    hidden = model.begin_state(func = mx.nd.zeros, batch_size = args_batch_size)
+    for i in range(0, data_source.shape[0] - 1, args_bptt):
+        data, target = get_batch(data_source, i)
+        data = data.as_in_context(ctx_begin)
+        target = target.as_in_context(ctx_end)
+        output, hidden = model.forward(data, hidden)
+        L = loss(output, target)
+        total_L += mx.nd.sum(L).asscalar()
+        ntotal += L.size
+    return total_L / ntotal
+```
+
+
+```python
+def train():
+    for epoch in range(args_epochs):
+        total_L = 0.0
+        start_time = time.time()
+        hidden = model.begin_state(func = mx.nd.zeros, batch_size = args_batch_size)
+        for ibatch, i in enumerate(range(0, train_data.shape[0] - 1, args_bptt)):
+            data, target = get_batch(train_data, i)
+            data = data.as_in_context(ctx_begin)
+            target = target.as_in_context(ctx_end)
+            hidden = detach(hidden)
+            with autograd.record():
+                output, hidden = model.forward(data, hidden)
+                L = loss(output, target)
+                L.backward()
+
+            model.clip_global_norm(args_clip * args_bptt * args_batch_size)
+
+            model.step(args_batch_size)
+            total_L += mx.nd.sum(L).asscalar()
+
+            if ibatch % args_log_interval == 0 and ibatch > 0:
+                cur_L = total_L / args_bptt / args_batch_size / args_log_interval
+                print('[Epoch %d Batch %d] loss %.2f, perplexity %.2f' % (
+                    epoch + 1, ibatch, cur_L, math.exp(cur_L)))
+                total_L = 0.0
+
+        val_L = eval(val_data)
+
+        print('[Epoch %d] time cost %.2fs, validation loss %.2f, validation perplexity %.2f' % (
+            epoch + 1, time.time() - start_time, val_L, math.exp(val_L)))
+
+```
+
+
+```python
+train()
+```
+
+    [Epoch 1 Batch 100] loss 7.20, perplexity 1340.10
```

Comment:
you can replace this with, that will allow your generated notebook to not have all this output:

```
`[Epoch 1 Batch 100] loss 7.20, perplexity 1340.10` <!--notebook-skip-line-->

`[Epoch 1 Batch 200] loss 6.66, perplexity 781.10` <!--notebook-skip-line-->

`...` <!--notebook-skip-line-->

`[Epoch 1 Batch 4800] loss 5.59, perplexity 267.57` <!--notebook-skip-line-->

`[Epoch 1] time cost 267.24s, validation loss 5.63, validation perplexity 279.41` <!--notebook-skip-line-->
```

Reply:
Thanks for the tip.